src/game/server/entities/character.cpp

```diff
@@ -2777,18 +2849,23 @@ bool CCharacter::TakeDamage(vec2 Force, int Dmg, int From, int Weapon, int Mode)
 	if(m_Health <= 0)
 	{
 		Die(From, Weapon);
+		
+		//KillerChar
+		CCharacter *pKillerChar = pKillerPlayer->GetCharacter();
 
 		// set attacker's face to happy (taunt!)
-		if (From >= 0 && From != m_pPlayer->GetCID() && GameServer()->m_apPlayers[From])
+		if (From >= 0 && From != m_pPlayer->GetCID())
 		{
-			CCharacter *pChr = GameServer()->m_apPlayers[From]->GetCharacter();
-			if (pChr)
+			if (pKillerChar)
 			{
-				pChr->m_EmoteType = EMOTE_HAPPY;
-				pChr->m_EmoteStop = Server()->Tick() + Server()->TickSpeed();
+				pKillerChar->m_EmoteType = EMOTE_HAPPY;
+				pKillerChar->m_EmoteStop = Server()->Tick() + Server()->TickSpeed();
 			}
 		}
-
+		
+		pKillerPlayer->IncreaseNumberKills();
+		pKillerChar->CheckSuperWeaponAccess();
```

Comment:
Server crashed here because pKillerChar was null.

I am pretty surprised with it, because we already check for GetCharacter() being non-null before.
But what i think happens is pKillerPlayer dies somewhere between the check and this moment. When it dies, its GetCharacter() on line 2854 returns 0.

I see 2 ways how to fix it. The first one is to check here the value of pKillerChar on non-null. The second is to call initialize pKillerChat earlier, on line 2699, for example.

Reply:
Yes it is weird, ok lets try the second method